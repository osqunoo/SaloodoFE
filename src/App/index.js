import React, { Component, Suspense } from 'react';
import { Switch, Route,Redirect  } from 'react-router-dom';
import Loadable from 'react-loadable';
import '../../node_modules/font-awesome/scss/font-awesome.scss';
import '../style.css';
//import 'font-awesome/scss/font-awesome.scss';
import Loader from './layout/Loader'
import Aux from "../hoc/_Aux";
import ScrollToTop from './layout/ScrollToTop';
import routes from "../route";
import cookie from 'react-cookies';
 import ManagerDashboard from '../components/Manager/Default';
 //const ManagerDashboard = React.lazy(() => import('../components/Manager/Default'));
const BikerDashboard = React.lazy(() => import('../components/Biker/Default'));
const Login = React.lazy(() => import('../components/Authentication/SignIn/Login')); 

const AdminLayout = Loadable({
    loader: () => import('./layout/AdminLayout'),
    loading: Loader
});


class App extends Component {
  render() {
    const menu = routes.map((route, index) => {
      return (route.component) ? (
          <Route
              key={index}
              path={route.path}
              exact={route.exact}
              name={route.name}
              render={props => (
                  <route.component {...props} />
              )} />
      ) : (null);
    });

    return (
        <Aux>
            <ScrollToTop>
                <Suspense fallback={<Loader/>}>
                    <Switch>
                        {menu}
                        <Route path="/" component={AdminLayout} />
                    </Switch>
                </Suspense>
            </ScrollToTop>
        </Aux>
    );
}
}

export default App;
