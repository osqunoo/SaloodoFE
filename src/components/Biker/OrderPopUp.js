import React from 'react';
import { Row, Col, Button, Modal, Form } from 'react-bootstrap';
import { connect } from 'react-redux'
import DatePicker from 'react-datepicker';
import Aux from '../../hoc/_Aux';
import { updateOrderAction, toggleOrderPopUp } from '../../actions/index';
import ORDER_STATUS from '../../constants/orderStatus';
import { ShipmentsServices } from '../../services/Shipments';
import 'react-datepicker/dist/react-datepicker.css';
import 'bootstrap/dist/css/bootstrap.min.css';
class OrderPopUp extends React.Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handlePickDateChange = this.handlePickDateChange.bind(this);
    this.handleDeliveredChange = this.handleDeliveredChange.bind(this);
    this.state = { ...this.state }
  }

  handlePickDateChange(date) {
    this.setState({ PICKED_UP_DATE: date });
  } 
  
  handleDeliveredChange(date) {
    this.setState({ EXPECTED_DELIVER_DATE: date });
  }
  handleSubmit(event) {
    event.preventDefault();
    var PICKED_UP_DATE= this.state.PICKED_UP_DATE != '' ? this.state.PICKED_UP_DATE.toUTCString() : '';
    var EXPECTED_DELIVER_DATE= this.state.EXPECTED_DELIVER_DATE != '' ? this.state.EXPECTED_DELIVER_DATE.toUTCString() : '';
    var shipmentSelected= this.props.shipmentSelected;
    ShipmentsServices.updateOrder(shipmentSelected,this.props.BIKERID,ORDER_STATUS.PICKED_UP,PICKED_UP_DATE,EXPECTED_DELIVER_DATE )
      .then((res) => 
      {
        this.props.updateOrderAction(res.data);
        this.props.toggleOrderPopUp();
      });
  }
       
  render() {
    return (
            <Aux>
                <Row>
                    <Col md={12} xl={12}>
                      <Modal show={this.props.OrderOpen} onHide={this.props.toggleOrderPopUp}>
                        <Modal.Header closeButton>
                          <Modal.Title>PickUp Order Timestamp</Modal.Title>
                        </Modal.Header>
                        <Form  onSubmit={this.handleSubmit}  >
                        <Modal.Body>
                          <h4>Pick Up</h4>
                          <p>If you pickup this order, Please select the pickup & expected Timestamp</p>
                          <Form.Group as={Row} controlId="PickUpTimestamp">
                              <Form.Label column sm="6" md="6" xl="6">PickUp Timestamp </Form.Label>
                              <DatePicker sm="6" md="6" xl="6"
                                selected={ this.state.PICKED_UP_DATE }
                                onChange={ this.handlePickDateChange }
                                name="PICKED_UP_DATE" 
                                showTimeSelect
                                timeFormat="HH:mm"
                                timeIntervals={15}
                                timeCaption="time"
                                dateFormat="MMMM d, yyyy h:mm aa"
                                required
                              />
                          </Form.Group> 
                          <Form.Group as={Row} controlId="DeliveredTimestamp">
                              <Form.Label column sm="6">Delivered Timestamp </Form.Label>
                              <DatePicker sm="6"
                                selected={ this.state.EXPECTED_DELIVER_DATE }
                                onChange={ this.handleDeliveredChange }
                                name="EXPECTED_DELIVER_DATE"
                                showTimeSelect
                                timeFormat="HH:mm"
                                timeIntervals={15}
                                timeCaption="time"
                                dateFormat="MMMM d, yyyy h:mm aa"
                                required
                              />
                          </Form.Group> 
                        </Modal.Body>
                        <Modal.Footer>
                          <Button type="submit" variant='success'>PickUp Order </Button>
                          <Button variant='danger' onClick={this.props.toggleOrderPopUp}>Close</Button>
                        </Modal.Footer>
                        </Form>
                      </Modal>
                    </Col>
                </Row>
            </Aux>
    );
  }
}
function mapDispatchToProps(dispatch) {
  return {
    updateOrderAction: (obj) => dispatch(updateOrderAction(obj)),
    toggleOrderPopUp: (order) => dispatch(toggleOrderPopUp(order)),
  };
}
function mapStateToProps(state) {
  return { 
    BIKERID: state.BIKERID,
    BIKERS: state.AllBikers,
    shipments: state.AllOrders,
    shipmentSelected: state.shipmentSelected,
    OrderOpen: state.OrderOpen,
    BikesSelected:state.BikesSelected
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderPopUp);