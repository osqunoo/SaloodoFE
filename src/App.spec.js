import React from 'react';
import { shallow } from 'enzyme';
import App from './App/index';
import Dashboard from './components/Manager/Default';
import Assign from './components/Manager/AssignPopUp';
import BikerDashboard from './components/Biker/Default';
import OrderPopUp from './components/Biker/OrderPopUp';
import { createMemoryHistory } from "history";

describe('App', () => {
     let wrapper;
     beforeEach(() => wrapper = shallow(<App />));
   it('should base path is /', () => {
     const history = createMemoryHistory();
     expect(history.location.pathname).toBe("/");
   });
 
 });
describe('Dashboard', () => {

  it("Manager Dashboard Component should render correctly", () => {
    const component = shallow(<Dashboard />);
  });
  it("Manager Dashboard Component should render initial layout", () => {
    const component = shallow(<Dashboard />);
    expect(component.getElements()).toMatchSnapshot();
  });
 
});
describe('Assign', () => {
    it("Assign Order Component should render correctly", () => {
        const component = shallow(<Assign />);
    });
    it("Assign Order Component should render initial layout", () => {
        const component = shallow(<Assign />);
        expect(component.getElements()).toMatchSnapshot();
    });
});

describe('BikerDashboard', () => {
    it("Biker Dashboard Component should render correctly", () => {
      const component = shallow(<BikerDashboard />);
    });
    it("Biker Dashboard Component should render initial layout", () => {
      const component = shallow(<BikerDashboard />);
      expect(component.getElements()).toMatchSnapshot();
    });
});

describe('OrderPopUp', () => {
    it("Update Order Component should render correctly", () => {
      const component = shallow(<OrderPopUp />);
    });
    it("Update Order Component should render initial layout", () => {
      const component = shallow(<OrderPopUp />);
      expect(component.getElements()).toMatchSnapshot();
    });
});