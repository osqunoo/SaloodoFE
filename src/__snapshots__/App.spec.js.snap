// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assign Assign Order Component should render initial layout 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;

exports[`BikerDashboard Biker Dashboard Component should render initial layout 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;

exports[`Dashboard Manager Dashboard Component should render initial layout 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;

exports[`Dashboard should render initial layout 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;

exports[`OrderPopUp Update Order Component should render initial layout 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;
